### PR TITLE
feat!: improve documentation for associated data

### PR DESCRIPTION
**Pull request by paolodamico**
- Introduces an explicit `id` to the Credential.
- Clarifies the use of associated data.
- Minor refactor on hashing utilities (location of functions, no logical changes).

---

> [!NOTE]
> Adds a random `id` to `Credential` and moves associated-data hashing to `sponge::hash_bytes_to_field_element` with a dedicated DS tag, tightening internal visibilities and tests.
> 
> - **Primitives**:
>   - **`Credential`**:
>     - Add `id: u64` with random default and `id()` builder.
>     - Use `sponge::hash_bytes_to_field_element(ASSOCIATED_DATA_HASH_DS_TAG, ...)` for `associated_data`; expand associated-data docs/comments.
>   - **`sponge` module**:
>     - Introduce public `hash_bytes_to_field_element(ds_tag, data)`; keep `hash_bytes_with_poseidon2_t16_r15` and helpers internal; make constants private.
>     - Add comprehensive hashing tests (determinism, size boundaries, errors); migrate related tests from `credential`.
>   - **Dependencies**:
>     - Add `rand` (workspace) for `Credential` ID generation.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 1bba9dad159d8642ae7180897f670ed9ba29612b. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: crates/primitives/src/credential.rs [hunk 1]
```diff
@@ -50,33 +81,56 @@ pub enum CredentialVersion {
...
+    /// A reference identifier for the credential. This can be used by issuers
+    /// to manage credential lifecycle.
+    ///
+    /// - This ID is never exposed or used outside of issuer scope. It is never part of proofs
+    ///   or exposed to RPs.
+    /// - Generally, it is recommended to maintain the default of a random identifier.
+    ///
+    /// # Example Uses
+    /// - Track issued credentials to later support revocation after refreshing.
+    pub id: u64,
```
[the remark this answers is not in the document]
Comment (paolodamico) [reply crates/primitives/src/credential.rs]: @dkales how feasible would it be to incorporate this with the circuit updates?

[the remark this answers is not in the document]
Comment (paolodamico) [reply crates/primitives/src/credential.rs]: yes, as in can we add it to the hash of the credential and reflect it in the circuit without it being too burdensome?

[the remark this answers is not in the document]
Comment (dkales) [reply crates/primitives/src/credential.rs]: No right now is the time to do that, i would see a few options:
* Put it in the associated data, no changes needed in the circuit
* Put it in the top-level hash, need to decide if it is a private or public input (but I would see private). However, I do not see what benefit this would have TBH compared to the associated data hash.

[the remark this answers is not in the document]
Comment (paolodamico) [reply crates/primitives/src/credential.rs]: let's do top level hash with a private input, we do have one empty slot in the permutation
